doc/mgmt/gnmi/gNMI_Subscription_for_YangData.md

```diff
@@ -0,0 +1,1183 @@
+  - [5.1 Finding Subscription Preferences for a YANG Path](#51-finding-subscription-preferences-for-a-yang-path)
+- [6 Scale Considerations](#6-scale-considerations)
+- [7 Limitations](#7-limitations)
+- [8 Unit Tests](#8-unit-tests)
+
+## Revision History
+
+| Rev | Date        |       Author       | Change Description                                                            |
+|-----|-------------|--------------------|-------------------------------------------------------------------------------|
+| 0.1 | 03/02/2023  | Sachin Holla       | Initial draft                                                                 |
+| 0.2 | 03/14/2023  | Balachandar Mani   | Added the CommonApp's interface, Transformer and Path validator details |
+
+## Definition and Abbreviation
+
+| **Term**       | **Meaning**                         |
+|----------------|-------------------------------------|
+| gNMI           | gRPC Network Management Interface   |
+| gRPC           | gRPC Remote Procedure Call (it is a recursive acronym!!)   |
+| RPC            | Remote Procedure Call.. In this document it mostly refers to the operations defined by gNMI specification |
+| YGOT           | [YANG Go Tools](https://github.com/openconfig/ygot); a Golang library to process YANG data |
+
+## 1 Feature Overview
+
+### 1.1 Introduction
+
+SONiC Telemetry service suports gNMI Get, Set and Subscribe RPCs for DB paths and sonic-yang based paths.
+It also suppots Get and Set for OpenConfig and IETF yang based paths that are part of **sonic-mgmt-common** repository.
+This design document describes proposed enhancements to support gNMI Subscribe RPC for such YANG paths.
+
+### 1.2 Requirements
+
+#### 1.2.1 ON_CHANGE subscription for eligible paths
+
+Infrastructure should support ON_CHANGE subscription for YANG paths based on redis keyspace notifications.
+It should provide APIs for apps to specify ON_CHANGE unsupported paths.
+An `INVALID_ARGUMENT` status should be returned if the target path or any of its descendent paths do not support ON_CHANGE.
+
+#### 1.2.2 SAMPLE subscription for all paths
+
+Infrastructure should support SAMPLE subscription for all YANG paths.
+It should provide APIs for apps to indicate minimum `sample_interval` supported for the path.
+An `INVALID_ARGUMENT` status should be returned if the minimum `sample_interval` of target path or any of its descendent paths is more than the requested `sample_interval`.
+
+#### 1.2.3 TARGET_DEFINED subscription for all paths
+
+Infrastructure should support TARGET_DEFINED subscription for all YANG paths.
+Subscribe request should be treated as ON_CHANGE or SAMPLE based on the app's preferences for the target path.
+It should split into multiple requests if the target path supports ON_CHANGE
+but some of its descendent paths do not.
+
+#### 1.2.4 POLL and ONCE subscriptions
+
+Infrastructure should support POLL and ONCE subscriptions for all YANG paths.
+
+#### 1.2.5 Support Wildcard Keys
```

Comment:
>Wildcard Keys [](http://example.com/codeflow?start=19&length=13)

wildcard is not limited to openconfig yang models. How about the support on sonic-yang paths, non-db paths, db paths?

Reply:
This HLD only covers the yang paths handled by translib.